lib/pages/chat/events/message_download_content.dart

```diff
@@ -165,7 +165,12 @@ class _MessageDownloadContentState extends State<MessageDownloadContent>
         }
 
         return InkWell(
-          onTap: () {
+          onTap: () async {
+            await checkFileInDownloadsInApp();
+            if (downloadFileStateNotifier.value
+                is DownloadedPresentationState) {
+              return;
+            }
```

Comment:
still not understand this logic

Reply:
- This logic handle the case when user download the file when click to `Save to downloads`, then tap to the file downloaded again
https://github.com/linagora/twake-on-matrix/assets/43041967/0300a87b-31ef-4158-b4ba-a4bbeb078dd1